src/gap.c

```diff
@@ -170,19 +170,16 @@ void ViewObjHandler ( Obj obj )
 
   /* if non-zero use this function, otherwise use `PrintObj'             */
   memcpy( readJmpError, TLS(ReadJmpError), sizeof(syJmp_buf) );
-  if ( ! READ_ERROR() ) {
+  TRY_READ {
```

Comment:
I'm not sure why this is here, but it's purpose is to make sure if a break loop is triggered while printing from the eval prompt, then we just drop back to the prompt, rather than longJmp back out of the current loop (which would cause an exit if it happened at the `gap>` prompt).

It took me in general time to realise that `READ_ERROR` is how we handle all errors, both invalid input, and ctrl+c / exiting break loops.